binaries/ofs.uio/uio/ofs.uio.py

```diff
@@ -0,0 +1,729 @@
+# * Neither the name  of Intel Corporation  nor the names of its contributors
+#   may be used to  endorse or promote  products derived  from this  software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+# IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+# LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+# CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+# SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+# INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+# CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import re
+import os
+import glob
+import argparse
+import sys
+import time
+import mmap
+import binascii
+import struct
+from ctypes import c_uint64, Structure, Union, c_uint32
+
+MAPSIZE = mmap.PAGESIZE
+MAPMASK = MAPSIZE-1
+
+PATTERN = (r'.*(?P<segment>\w{4}):(?P<bus>\w{2}):'
+           r'(?P<dev>\w{2})\.(?P<func>\d).*')
+
+FPGA_ROOT_PATH = '/sys/class/fpga_region'
+
+BDF_PATTERN = re.compile(PATTERN, re.IGNORECASE)
+
+DEFAULT_BDF = 'ssss:bb:dd.f'
+
+# PCIe SS Feature id and Mailbox csr offset
+PCIESS_FEATURE_ID = 0x20
+MAILBOX_CMD_CSR = 0x28
+MAILBOX_RD_DATA_CSR = 0x30
+MAILBOX_WR_DATA_CSR = 0x31
+
+# mailbox register poll interval 1 microseconds
+MAILBOX_POLL_SLEEP_TIME = 1/1000000
+
+# mailbox register poll timeout 100 microseconds
+MAILBOX_POLL_TIMEOUT = 1/10000
+
+
+def verify_pcie_address(pcie_address):
+    m = BDF_PATTERN.match(pcie_address)
+    if m is None:
+        print("Invalid pcie address format{}".format(pcie_address))
+        return False
+    return True
+
+
+class verify_hex_list(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        regx = re.compile("0[xX][0-9a-fA-F]")
+        for value in values:
+            try:
+                if re.search(regx, value) or int(value, 16):
+                    continue
+                else:
+                    parser.error(f"Please enter a valid. Got: {values}")
+            except ValueError:
+                parser.error(f"Please enter a valid. Got: {values}")
+        setattr(namespace, self.dest, values)
+
+
+def verify_hex(str):
+    regx = re.compile("0[xX][0-9a-fA-F]")
+    try:
+        if re.search(regx, str) or int(str, 16):
+            return int(str, 16)
+        else:
+            raise argparse.ArgumentTypeError('Invalid input:', str)
+    except ValueError:
+        raise argparse.ArgumentTypeError('Invalid input:', str)
+
+
+def verify_uio(str):
+    regx = re.compile("uio[0-9]")
```

Comment:
This would only allow 1-digit uio devices.  When the uio10 is created, this will break.

Reply:
fixed